lib/os/__init__.py

```diff
@@ -139,7 +139,7 @@ def close(self, _):
     result = _encode_wait_result(state.Sys())
     if not result:
       return None
-    return state.Pid(), result
+    return state.status 
```

Comment:
I think you should return `result` here, which should be the same as state.status on most platforms.